platform_versioned_docs/version-24.1/compute-envs/azure-batch.mdx

```diff
@@ -117,7 +121,37 @@ After you have created a resource group and storage account, create a [Batch acc
+
+:::note
+The two Azure credential types use entirely different authentication methods. You can add more than one credential to a workspace, but only one can be used at a time. While they can be used concurrently, they are not cross-compatible, and access granted by one will not be shared with the other.
+:::
+
+#### Access Keys
+
+1. Navigate to the Azure Portal and sign in.
+1. Locate the Azure Batch account and select "Keys" under "Account management." Here, you will see the Primary and Secondary keys. Copy one of the keys and save it in a secure location for later use.
+1. Locate the Azure Storage account and, under the "Security and Networking" section, select "Access keys". Here, you will see Key1 and Key2 options. Copy one of them and save it in a secure location for later use. Be sure to delete them after saving them in Seqera Platform.
+1. In Seqera Platform, go to your workspace, select "Add a new credential," and choose the "Azure Credentials" type.
+1. Enter a name for the credentials, such as _Azure Credentials_.
+1. Add the **Batch account** and **Blob Storage** account names and access keys.
+
+#### Service Principal
+
+1. In the Azure Portal, navigate to "Microsoft Entra ID," and under "App registrations," click "New registration." See the [Azure documentation][az-create-sp] for more details.
+1. Provide a name for the application. The application will automatically have a Service Principal associated with it.
+1. Assign roles to the Service Principal. Go to the Azure Storage account, and under "Access Control (IAM)," click "Add role assignment."
+1. Choose the roles "Storage Blob Data Reader" and "Storage Blob Data Contributor", then select "Members", click "Select Members", search for your newly created Service Principal, and assign the role.
+1. Repeat the same process for the Azure Batch account, but use the "Azure Batch Contributor" role.
+1. Seqera Platform will need credentials to authenticate as the Service Principal. Navigate back to the app registration, and on the "Overview" page, save the "Application (client) ID" value for use in Seqera Platform.
+1. Then, click "Certificates & secrets" and select "New client secret." A new secret will be created containing a value and secret ID. Save both of these securely for use in Seqera Platform. Be sure to delete them after saving them in Seqera Platform.
+1. In Seqera Platform, go to your workspace, select "Add a new credential," choose the "Azure Credentials" type, then select the "Service Principal" tab.
+1. Enter a name for the credentials, such as _Azure Credentials_.
+1. Add the Application ID, Secret ID, Secret, **Batch account**, and **Blob Storage** account names to the relevant fields.
+
+## Seqera Platform
 
 There are two ways to create an Azure Batch compute environment in Seqera Platform:
```

Comment:
Is this true? where are the two ways detailed?  
this is only true for one subtype of azure credentials. Service principals can only be used with manual CEs at the moment.

Reply:
Clarified.